File: src/tutorial/hello_object.cc
#include "tutorial/hello_object.hh"

#include <iostream>

HelloObject::HelloObject(HelloObjectParams *params) :
        SimObject(params)
{
        std::cout << "Hello World from SimObject!" << std::endl;
}

HelloObject *HelloObjectParams::create()
{
        return new HelloObject(this);
}

File: src/tutorial/HelloObject.py
from m5.params import *
from m5.SimObject import SimObject

class HelloObject(SimObject):
    type = 'HelloObject'
    cxx_header = "tutorial/hello_object.hh"

    time_to_wait = Param.Latency("Time before firing an event.")
    number_of_fires = Param.Int(1, "No. of times to fire
            events before goodbye.")

